### PR TITLE
#30296 Change Signature Dialog update to include VB style.

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
#30296 Update the code to include separate function for the different languages (C# and Visual Basic).

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs [hunk 1]
```diff
@@ -440,89 +440,117 @@ public string RestoreAutomationText
...
+
+                        if (ParameterSymbol.IsParams)
+                        {
+                            return "ParamArray";
+                        }
+
+                        if (_changeSignatureDialogViewModel._thisParameter != null &&
+                            ParameterSymbol == _changeSignatureDialogViewModel._thisParameter.ParameterSymbol)
+                        {
+                            return "Me";
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs]: this seems wrong...

File: src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs [hunk 2]
```diff
@@ -440,89 +440,117 @@ public string RestoreAutomationText
...
+                        }
+                        return string.Empty;
                     }
 
-                    if (_changeSignatureDialogViewModel._thisParameter != null &&
-                        _parameter == _changeSignatureDialogViewModel._thisParameter._parameter)
+                    switch (ParameterSymbol.Language)
                     {
-                        return "this";
+                        case "C#":
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs]: use LanguageNames

File: src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs [hunk 3]
```diff
@@ -440,89 +440,117 @@ public string RestoreAutomationText
...
                 get
                 {
-                    if (!_parameter.HasExplicitDefaultValue)
+                    string CS()
+                    {
+                        return ParameterSymbol.ExplicitDefaultValue == null
+                        ? "null"
+                        : ParameterSymbol.ExplicitDefaultValue is string
+                            ? "\"" + ParameterSymbol.ExplicitDefaultValue.ToString() + "\""
+                            : ParameterSymbol.ExplicitDefaultValue.ToString();
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs]: boy is this ick.  but i see that's how the original code as as well... :-/

File: src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs [hunk 4]
```diff
@@ -440,89 +440,117 @@ public string RestoreAutomationText
...
+                    }
+                    string VB()
+                    {
+                        return ParameterSymbol.ExplicitDefaultValue == null ? "Nothing" :
+                               ParameterSymbol.ExplicitDefaultValue.ToString();
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs]: not following IDE types (shown above) for `?:`.  Why not handle the string case like the C# version does?

File: src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs [hunk 5]
```diff
@@ -440,89 +440,117 @@ public string RestoreAutomationText
...
+                            return "params";
+                        }
+
+                        if (_changeSignatureDialogViewModel._thisParameter != null &&
+                            ParameterSymbol == _changeSignatureDialogViewModel._thisParameter.ParameterSymbol)
+                        {
+                            return "this";
+                        }
+                        return string.Empty;
                     }
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs]: not a fan of local functions here.  however, if you are going to use local functions, place at end of function after the final 'return' statement.

File: src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs [hunk 6]
```diff
@@ -440,89 +440,117 @@ public string RestoreAutomationText
...
-                        _parameter == _changeSignatureDialogViewModel._thisParameter._parameter)
+                    switch (ParameterSymbol.Language)
                     {
-                        return "this";
+                        case "C#":
+                            return CS();
+                        case "Visual Basic":
+                            return VB();
+                        default:
+                            return string.Empty;
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs]: i don't see any benefit to these local function which you then immediately invoke once.  why not just test the language and have the language specific code in the if-block?

File: src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs [hunk 7]
```diff
@@ -440,89 +440,117 @@ public string RestoreAutomationText
...
+                    if (!ParameterSymbol.HasExplicitDefaultValue)
                     {
                         return string.Empty;
                     }
+                    switch(ParameterSymbol.Language)
+                    {
+                        case "CS": return CS();
+                        case "Visual Basic": return VB();
+                        default: return string.Empty;
+                    }
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/ChangeSignature/ChangeSignatureDialogViewModel.cs]: all the same above complaints apply here.

**Comment by heejaechang** [comment]
retest windows_debug_vs-integration_prtest  please

**Comment by heejaechang** [comment]
thank you!

**Comment by heejaechang** [comment]
the failure is known issue - ApplyEditorConfigAndFixAllOccurrences

**Pull request merged**